### ISSUE TITLE
Build failure: python3Packages.fpdf2

### ISSUE THREAD

**Issue by josephst** [NixOS/nixpkgs#308941]
### Steps To Reproduce

1. build fpdf2 (confirmed on x64-linux and aarch64-darwin) - `nix-shell -p "python311.withPackages(ps: with ps; [ fpdf2 ])"`

### Build log

```
       > =========================== short test summary info ============================
       > FAILED test/image/image_types/test_insert_images.py::test_insert_jpg_jpxdecode - AssertionError: 54f2cbf1c077910aaed684fb686ef6dc != 16dfee1960e1e33fc9f6988...
       > FAILED test/text_shaping/test_bidirectional.py::test_bidi_conformance - urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] cert...
       > FAILED test/text_shaping/test_bidirectional.py::test_bidi_character - urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] cert...
       > = 3 failed, 1230 passed, 1 skipped, 3 deselected, 1 xfailed, 7 warnings in 12.30s =
```

### Additional context

The build failures for `test_bidirectional.py` are due to these tests [attempting to download a test file containing bidirectional test](https://github.com/py-pdf/fpdf2/blob/d7d0ab222d6c288b677e57201d6c95705f11c5c7/test/text_shaping/test_bidirectional.py#L67). As network access isn't allowed, these two tests fail. 

The failure of `test_insert_images.py` is more puzzling to me, I don't see any recent changes to these tests. Perhaps a missing dependency on Nix?

Right now I've got these tests disabled and the module builds & tests successfully again. 
```
      (fpdf2.overridePythonAttrs(old: {
        disabledTests = old.disabledTests ++ [
          "test_bidi_character" # tries to download file
          "test_bidi_conformance" # tries to download file
          "test_insert_jpg_jpxdecode" # JPEG2000 error?
        ];
      }))
```

### Notify maintainers
@jfvillablanca
### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.7, NixOS, 24.05 (Uakari), 24.05.20240428.cf8cc12`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.22.0`
 - nixpkgs: `/nix/store/4ab6vrcph07w6ra79bc04fy8bbcmb9r0-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
Last error is related to the `python3Packages.fonttools` upgrade. Reverting the last version bump fixes that test.

See https://github.com/py-pdf/fpdf2/issues/899

**Comment by FliegendeWurst** [comment]
Fixed in #309846
